core/server/master/src/main/java/alluxio/master/file/meta/UfsSyncPathCache.java

```diff
@@ -98,7 +98,11 @@ public UfsSyncPathCache(Clock clock) {
                 if (onRemoval != null) {
                   onRemoval.accept((String) removal.getKey(), (SyncState) removal.getValue());
                 }
-                onCacheEviction((String) removal.getKey(), (SyncState) removal.getValue());
+                // We decide to stop updating invalidation time for paths in order to avoid
+                // paths got evicted recursively. Stopping calling this onCacheEviction
+                // function will result in stopping updating invalidation time for all paths.
+                // Since no one is using this invalidation time feature, we decide to disable
+                // updating invalidation time.
```

Comment:
```suggestion
                // Due to issue https://github.com/Alluxio/alluxio/issues/blah,
                // we decide to stop updating invalidation time for paths in order to avoid
                // paths got evicted recursively. Stopping calling this onCacheEviction
                // function will result in stopping updating invalidation time for all paths.
                // Since no one is using this invalidation time feature, we decide to disable
                // updating invalidation time.
```